UnitySDK/Assets/ML-Agents/Scripts/Agent.cs

```diff
@@ -374,7 +373,9 @@ public void AddReward(float increment)
         /// <returns>The step reward.</returns>
         public float GetReward()
         {
-            return m_Reward;
+            Debug.Assert(m_LegacyRewardProvider != null, "LegacyRewardProvider is null and " +
+                "legacy method 'GetReward' was called.");
+            return m_LegacyRewardProvider.GetIncrementalReward();
```

Comment:
Calling GetIncrementalReward will reset the reward. A user calling GetReward might end up deleting it. I would remove this method or re-add the reward to the reward provider.

Reply:
yeah, I did that locally already.